src/workspaces/common/state/useWorkspaceStatePolling.ts

```diff
@@ -0,0 +1,118 @@
+import { LoadedState } from '@terra-ui-packages/core-utils';
+import _ from 'lodash/fp';
+import { useEffect, useState } from 'react';
+import { Ajax } from 'src/libs/ajax';
+import { workspacesStore, workspaceStore } from 'src/libs/state';
+import { pollWithCancellation } from 'src/libs/utils';
+import { WorkspaceState, WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+const updateWorkspacesList = (
+  workspaces: Workspace[],
+  workspaceId: string,
+  state: WorkspaceState,
+  errorMessage?: string
+): Workspace[] => workspaces.map((ws) => updateWorkspace(ws, workspaceId, state, errorMessage));
+
+const updateWorkspace = <T extends Workspace>(
+  workspace: T,
+  workspaceId: string,
```

Comment:
Suggest `maybeUpdateWorkspace` or something like that, since this actually only updates if `workspaceId` matches the `workspace`.